args/linkedin.go

```diff
@@ -10,6 +18,72 @@ type LinkedInArguments struct {
 	Start            int      `json:"start"`                     // Pagination start offset
 }
 
+// UnmarshalJSON implements custom JSON unmarshaling with validation
+func (l *LinkedInArguments) UnmarshalJSON(data []byte) error {
+	type Alias LinkedInArguments
+	aux := &struct {
+		*Alias
+	}{
+		Alias: (*Alias)(l),
+	}
+
+	if err := json.Unmarshal(data, aux); err != nil {
+		return fmt.Errorf("failed to unmarshal LinkedIn arguments: %w", err)
+	}
+
+	// Normalize QueryType to lowercase
+	if l.QueryType != "" {
+		l.QueryType = strings.ToLower(l.QueryType)
+	}
+
+	return l.Validate()
+}
+
+// Validate validates the LinkedIn arguments
+func (l *LinkedInArguments) Validate() error {
+	if l.QueryType == "" {
+		return fmt.Errorf("type is required")
+	}
+
+	// Validate query type
+	validTypes := map[string]bool{
+		"searchbyquery": true,
+		"getprofile":    true,
+	}
+	if !validTypes[l.QueryType] {
```

Comment:
Can we use a `Set`, and make it a global `var`? E.g.

```
var validTypes = util.NewSet("searchbyquery", "getprofile")
```

and then down here

```
if !validTypes.Contains(l.QueryType) {
```

We probably want to do this everywhere (for now, at least in the new code, we should cleanup later)